std/os/linux.zig

```diff
@@ -1607,6 +1607,32 @@ pub const dirent64 = extern struct {
     d_name: u8, // field address is the address of first byte of name https://github.com/ziglang/zig/issues/173
 };
 
+pub const SS_ONSTACK = 1;
+pub const SS_DISABLE = 2;
+pub const SS_AUTODISARM = 1 << 31;
+
+pub const SIGSTKSZ = switch (builtin.arch) {
+    .i386, .x86_64, .arm => 8192,
+    .aarch64 => 16384,
+    else => @compileError("SIGSTKSZ not defined for this architecture"),
+};
+
+pub const MINSIGSTKSZ = switch (builtin.arch) {
+    .i386, .x86_64, .arm => 2048,
+    .aarch64 => 6144,
+    else => @compileError("MINSIGSTKSZ not defined for this architecture"),
+};
+
+pub const stack_t = extern struct {
+    ss_sp: usize,
```

Comment:
Should this be a `[*]u8` or a `*c_void`?

Reply:
> Should this be a [*]u8 or a *c_void?

It's a matter of picking the poison you like the most, you either need a `@pointerToInt` or a `@ptrCast`.

Reply:
With a pointer type you can't accidentally pass a number. Also pointers have additional properties such as alignment.

Reply:
Oh well, some more type-safety is always welcome. I'll turn this into a *c_void for symmetry with the other structure definitions.